apps/utils/prompt.py

```diff
@@ -27,8 +28,16 @@ def validate_prompt_variables(form_data, prompt_key: str, known_vars: set):
     to be used, otherwise a `ValidationError` is thrown.
     """
     prompt_text = form_data.get(prompt_key, "")
+    if not prompt_text:
+        return set()
+
+    prompt_variables = set()
     try:
-        prompt_variables = {get_root_var(var) for var in PromptTemplate.from_template(prompt_text).input_variables}
+        for literal, field_name, format_spec, conversion in Formatter().parse(prompt_text):
+            if field_name is not None:
+                if format_spec:
```

Comment:
Just thinking that we should also disallow conversions:
```suggestion
                if format_spec or conversion:
```

Here's a test case for conversions: `"{var!s}"`, and one for both: `"{var!r:xyz}"`